resharper/resharper-unity/src/HlslSupport/Integration/Cpp/UnityGameEngineDirectiveResolver.cs

```diff
@@ -51,85 +55,154 @@ public string TransformPath(CppInclusionContext context, string path)
-                return localPackagePath.FullPath;
 
-            var cachedPackagePath = FileSystemPath.Parse("Library").Combine("PackageCache")
-                .Combine(packageName + "@" + suffix + path.Substring(endPos));
-            return cachedPackagePath.FullPath;
+            return packagePath + path.Substring(endPos);
         }
-
-        [CanBeNull]
-        private string GetCacheSuffix(string packageName)
+        
+        private void ParsePackageLock()
         {
-            // TODO: This cache should be based on resolved packages, like we have in the frontend for PackageManager
-            // TODO: Invalidate this cache when Packages/manifest.json or Packages/packages-lock.json changes
-            if (myCacheSuffixes == null)
+            try
             {
-                lock (myLockObject)
+                var packagesLockPath = mySolution.SolutionDirectory.Combine("Packages").Combine("packages-lock.json");
+                if (packagesLockPath.Exists == FileSystemPath.Existence.File)
                 {
-                    if (myCacheSuffixes == null)
+                    var json = JObject.Parse(packagesLockPath.ReadAllText2().Text);
+                    var dependencies = json["dependencies"]?.AsArray();
+                    if (dependencies != null)
                     {
-                        var result = new ConcurrentDictionary<string, string>();
-                        try
+                        foreach (var dependency in dependencies)
                         {
-                            var solutionFolder = mySolution.SolutionDirectory;
-                            var packagesFolder = solutionFolder.Combine("Packages");
-                            var packagesCacheFolder = solutionFolder.Combine("Library").Combine("PackageCache");
+                            if (!(dependency is JProperty jProperty))
+                                continue;
+                            var packageName = jProperty.Name;
+                            var source = (jProperty.Value["source"] as JValue)?.Value as string;
+                            var version = (jProperty.Value["version"] as JValue)?.Value as string;
+                            if (source == null || version == null)
+                                continue;
 
-                            var candidates = new[] {packagesFolder, packagesCacheFolder};
-                            foreach (var candidate in candidates)
+                            if (source.Equals("embedded"))
                             {
-                                foreach (var folder in candidate.GetDirectoryEntries())
+                                var packagePath = FileSystemPath.TryParse(version.RemoveStart("file:"));
+                                if (packagePath.IsEmpty)
+                                    continue;
+
+                                if (packagePath.IsAbsolute)
```

Comment:
Doesn't need changing, but for embedded packages, this will always be a relative path inside `Packages`, and Unity only recognises single level folders, so this will always be a single folder name. E.g. `file:com.unity.whatever`